File: Students/Xin_Jin/lc26.java
package BootCamp;

public class lc26 {
    public static void main(String[] args) {
        System.out.println(removeDuplicates(new int[]{3, 2, 3, 6, 3, 10, 9, 3}));
    }
    public static int removeDuplicates(int[] nums) {
        int j =0 ;
        for(int i=0;i<nums.length-1;i++)
        {
            if(nums[i]!= nums[i+1])
            {
                nums[j+1]=nums[i+1];
                j=j+1;
            }
        }
        return j+1;
    }
}

File: Students/JacobFlickinger/TwoPointers/lc26.js
var removeDuplicates = function(nums) {
    for (let i = 0; i < nums.length; i++) {
        let j = i + 1;
        while (nums[j] == nums[i]) {
            j++;
        }
        if (j == nums.length) {
            break;
        }
        let diff = j - 1 - i;
        for (let x = j; x < nums.length; x++) {
            nums[x - diff] = nums[x];
        }
    }
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] == nums[nums.length - 1]) {
            return i + 1;
        }
    }
};
